migrations/dimensions-config/1fae088c80b6.py

```diff
@@ -0,0 +1,877 @@
+"""Migration script for dimensions.yaml namespace=daf_butler version=6.
+
+Revision ID: 1fae088c80b6
+Revises: 2a8a32e1bec3
+Create Date: 2024-03-12 14:35:38.888572
+
+"""
+
+from __future__ import annotations
+
+import logging
+from collections.abc import Iterator
+from typing import Any, TypeAlias
+
+import alembic
+import sqlalchemy as sa
+from alembic import op
+from lsst.daf.butler import Timespan
+from lsst.daf.butler_migrate.butler_attributes import ButlerAttributes
+from lsst.utils import doImportType
+from sqlalchemy.dialects.postgresql import INT8RANGE, Range
+
+# revision identifiers, used by Alembic.
+revision = "1fae088c80b6"
+down_revision = "2a8a32e1bec3"
+branch_labels = None
+depends_on = None
+
+# Logger name should start with lsst to work with butler logging option.
+_LOG = logging.getLogger(f"lsst.{__name__}")
+
+
+def upgrade() -> None:
+    """Perform schema upgrade."""
+    ctx = _Context()
+    _lock_exposure_table(ctx)
+    _migrate_day_obs(ctx)
+    _migrate_groups(ctx)
+    _migrate_dimensions_json(ctx)
+
+
+def downgrade() -> None:
+    """Perform schema downgrade."""
+    raise NotImplementedError()
+
+
+def _lock_exposure_table(ctx: _Context) -> None:
+    # In this migration we generate new tables based on the content of the
+    # exposure table, so make sure that it is not modified while we are
+    # working.
+
+    if ctx.is_sqlite:
+        # Sqlite does not support table locks
+        return
+
+    _LOG.info("Locking exposure table")
+    schema = ""
+    if ctx.schema:
+        schema = f"{ctx.schema}."
```

Comment:
I'm also doing this interpolation and hope that no one is using fancy schema names, but maybe we should do proper quoting with `IdentifierPreparer`. (on some later ticket)